gusto/forcing.py

```diff
@@ -221,212 +220,91 @@ def _build_forcing_solver(self):
 
     def apply(self, scaling, x_in, x_nl, x_out, **kwargs):
 
-        self.x0.assign(x_nl)
-        self.scaling.assign(scaling)
-        if 'mu_alpha' in kwargs and kwargs['mu_alpha'] is not None:
-            self.mu_scaling.assign(kwargs['mu_alpha'])
-        self.u_forcing_solver.solve()  # places forcing in self.uF
-
-        u_out, p_out, _ = x_out.split()
-
-        x_out.assign(x_in)
-        u_out += self.uF
-
+        super(IncompressibleForcing, self).apply(scaling, x_in, x_nl, x_out, **kwargs)
         if 'incompressible' in kwargs and kwargs['incompressible']:
+            _, p_out, _ = self.x_out.split()
```

Comment:
Yeah, this looks like action at a distance.  You apply to `x_out`, but then do some extra nonsense to `self.x_out`.  I think you should just use `x_out` here.